controllers/databaseclusterrestore_controller.go

```diff
@@ -166,6 +175,28 @@ func (r *DatabaseClusterRestoreReconciler) Reconcile(ctx context.Context, req ct
 	return ctrl.Result{}, nil
 }
 
+func (r *DatabaseClusterRestoreReconciler) ensureOwnerReference(
+	ctx context.Context,
+	restore *everestv1alpha1.DatabaseClusterRestore,
+	db *everestv1alpha1.DatabaseCluster,
+) error {
+	if len(restore.GetOwnerReferences()) == 0 {
+		restore.SetOwnerReferences([]metav1.OwnerReference{
+			{
+				APIVersion:         db.APIVersion,
+				Kind:               db.Kind,
+				Name:               db.Name,
+				UID:                db.UID,
+				BlockOwnerDeletion: pointer.ToBool(true),
```

Comment:
A bit worried about this blocking. It would work fine when we'll have the `deleting` status and won't actually delete the cluster until all related backups and restores are deleted, but if we merge it now wouldn't it block the DB deletion for an non-obvious for the end user reason? I may miss something, but I mean - I try to delete the cluster, but it's not getting deleted bc there are still a restore exist, so I should delete the restore first. Wouldn't it be confusing?

Reply:
The DBC deletion would only get blocked if it failed to delete the DBR but I don't see a reason why it would fail to delete the DBR, there is nothing that would block the deletion of the DBR, e.g. there are no finalizers.

Reply:
Ok, I see, looks like I've misinterpret the field. Resolved.